altapay.php

```diff
@@ -3041,11 +3042,22 @@ public function createTransaction($savecard,
 
             return $response;
         }
+
+        $languageId = $this->context->language->id;
+        $languageCode = $this->context->language->iso_code;
+        $shopId = $this->context->shop->id;
+
+        if ($parent_order) {
+            $shopId = $parent_order->id_shop;
+            $languageId = $parent_order->getCustomer()->id_lang;
+            $languageCode = Db::getInstance()->getValue('SELECT iso_code FROM ' . _DB_PREFIX_ . 'lang WHERE `id_lang` = ' . (int) $languageId);
```

Comment:
You are casting a string into int. Please understand that an empty string will be casted as 0, which is a language that might exists.

![image](https://github.com/user-attachments/assets/805d6ee6-6e6e-45bb-981c-c75204df459b)

Reply:
Updated!